src/nvim/decoration.c

```diff
@@ -520,7 +524,7 @@ static void decor_range_insert(DecorState *state, DecorRange range)
 }
 
 void decor_range_add_virt(DecorState *state, int start_row, int start_col, int end_row, int end_col,
-                          DecorVirtText *vt, bool owned)
+                          DecorVirtText *vt, bool owned, DecorPriority subpriority)
```

Comment:
Added an argument to set `subpriority` so that we only need to add a new field to `DecorRange`. Alternatively we could add `subpriority` fields to both `DecorVirtText` and `DecorSignHighlight` and leave the signatures of `decor_range_add_virt` and `decor_range_add_sh` unchanged.